### PR TITLE
cors issues fixed and state username color bug fixed

### LINKED ISSUE #127: CORS error
## Expected Behavior

All the API's call should be working.

## Current Behavior and Triggers

Many features are blocked due to CORS error in API call.

## Possible Solution

I tried many soln, it did'nt work at all, i also tried stackoverflow soln but it also did'nt  worked.
Adding this link https://cors-anywhere.herokuapp.com/ before all the api links will work but it will make app too slow.
Can you suggest any other soln.

## Would you like to work on this issue?

Yes

## Screenshot of issue

Screenshot (117)

### PR DESCRIPTION
**Pull request by prathmeshkhandelwal1**
Fixes #127 

### Description

CORS added and bug fixed.

### Screenshot After Change:

Screenshot (124)

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### Non Functional Requirement
- [x] Follows the code style of this project.
- [x] I have performed a self-review of my own code
- [x] All new and existing tests passed.

### CODE REVIEW

File: public/client.js [hunk 1]
```diff
@@ -24,11 +24,12 @@ $('#username').keyup(function () {
   const username = $('#username')
   if (username.val()) {
     // https://aashutoshrathi.glitch.me/api/gh/ to increase API calls without 403.
-    const profile = 'https://aashutoshrathi.glitch.me/api/gh/' + username.val()
+    const profile = 'https://cors-anywhere.herokuapp.com/https://api.aashutosh.dev/gh/' + username.val()
```
Comment (aashutoshrathi) [inline public/client.js:27]: No need of CORS anywhere here, I have whitelisted `getmein.glitch.me`

File: public/client.js [hunk 2]
```diff
@@ -84,7 +85,7 @@ $(function () {
     event.preventDefault()
     const username = $('#username').val()
     const email = $('#email').val()
-    fetch(`https://api.github.com/search/users?q=${username}`)
+    fetch(`https://cors-anywhere.herokuapp.com/https://api.github.com/search/users?q=${username}`)
```
Comment (aashutoshrathi) [inline public/client.js:88]: Same here

**Review by aashutoshrathi — CHANGES_REQUESTED** [review]
Also, @iiitv/project-maintainers who added jQuery to this project? 😭

**Comment by AmanRaj1608** [comment]
I thought we were using jQuery from the [start](https://github.com/iiitv/getmein-web/blob/1eec74b15f9d34f0b2131e58f70bb64be98f8137/views/index.html). 🤔
However, we can now move from jQuery to vanilla JS.

Also, I was thinking if we can have a new design 😅

**Comment by aashutoshrathi** [comment]
Oh! Got it. jQuery was added for UIKit but not for us 😟

File: public/client.js (hunk 1 above)
Comment (prathmeshkhandelwal1) [reply public/client.js:27]: Screenshot (137)


> No need of CORS anywhere here, I have whitelisted `getmein.glitch.me`

But it is still giving CORS error

Comment (prathmeshkhandelwal1) [reply public/client.js:27]: I also tried some other URLs but the results were same :(
Screenshot (135)
Screenshot (136)

Comment (aashutoshrathi) [reply public/client.js:27]: Bro you are testing on localhost, I'm saying I whitelisted GetMeIn.glitch.me

**Review by aashutoshrathi — CHANGES_REQUESTED** [review]
Check comment

**Comment by aashutoshrathi** [comment]
How about we move API url to env so that if URL changes we don't have to change in code.

**Comment by prathmeshkhandelwal1** [comment]
> How about we move API url to env so that if URL changes we don't have to change in code.

That's a great idea Sir, will implement in text commit today.

**Comment by aashutoshrathi** [comment]
Also, update this [Table](https://github.com/iiitv/getmein-web#environment-variables) in README

**Comment by prathmeshkhandelwal1** [comment]
> Also, update this [Table](https://github.com/iiitv/getmein-web#environment-variables) in README

Sure sir

**Comment by prathmeshkhandelwal1** [comment]
> How about we move API url to env so that if URL changes we don't have to change in code.

Sir, as there is jquery in client.js, i am little bit confused about how to do it. i created the env file and also used  `require('dotenv').config()` in client file but it is giving error. i also tried to find soln online but it does'nt help. can you plz help me with this.

**Comment by aashutoshrathi** [comment]
We are already using env vars at many places in the project, just read and understand the flow there.
Also, what is the error you are getting?

File: public/client.js [hunk 3]
```diff
@@ -23,12 +23,12 @@ var validateEmail = function (emailValue) {
 $('#username').keyup(function () {
   const username = $('#username')
   if (username.val()) {
-    // https://aashutoshrathi.glitch.me/api/gh/ to increase API calls without 403.
-    const profile = 'https://aashutoshrathi.glitch.me/api/gh/' + username.val()
+    const profile = 'https://api.aashutosh.dev/gh/' + username.val()
```
Comment (aashutoshrathi) [inline public/client.js:26]: Use Template Strings

File: public/client.js [hunk 4]
```diff
@@ -23,12 +23,12 @@ var validateEmail = function (emailValue) {
...
     fetch(profile)
       .then(response => response.json())
       .then(data => {
-        if (data.message) {
+        console.log(data)
```
Comment (aashutoshrathi) [inline public/client.js:30]: And remove this log

**Review by aashutoshrathi — CHANGES_REQUESTED** [review]

**Comment by prathmeshkhandelwal1** [comment]
> We are already using env vars at many places in the project, just read and understand the flow there.
> Also, what is the error you are getting?

while using `require('dotenv').config()` in client.js ..its giving require is not defined

**Comment by prathmeshkhandelwal1** [comment]
Sir i committed rest changes in this commit other than env file.

**Comment by aashutoshrathi** [comment]
@prathmeshkhandelwal1 Got it, it is client-side JS, so just do the requested changes and I guess we are good to go.

File: public/client.js [hunk 5]
```diff
@@ -23,11 +23,10 @@ var validateEmail = function (emailValue) {
 $('#username').keyup(function () {
   const username = $('#username')
   if (username.val()) {
-    const profile = 'https://api.aashutosh.dev/gh/' + username.val()
+    const profile = `https://api.aashutosh.dev/gh/` + username.val()
```
Comment (aashutoshrathi) [inline public/client.js:26]: Add the variable in the template string itself.

**Comment by prathmeshkhandelwal1** [comment]
@aashutoshrathi Can u review this again?

**Review by aashutoshrathi — APPROVED** [review]
LGTM

**Comment by aashutoshrathi** [comment]
Will try to deploy and test this tommorow.

**Comment by aashutoshrathi** [comment]
@iiitv/project-maintainers Can you deploy and check this?

**Pull request merged**